tests/unit_tests/test_retry.py

```diff
@@ -1,18 +1,28 @@
 """retry tests"""
 
+import asyncio
 import dataclasses
 import datetime
+import sys
 from typing import Iterator
 from unittest import mock
 
 import pytest
 from wandb.sdk.lib import retry
 
+if sys.version_info >= (3, 10):
+    asyncio_run = asyncio.run
+else:
+
+    def asyncio_run(coro):
+        return asyncio.new_event_loop().run_until_complete(coro)
+
 
 @dataclasses.dataclass
 class MockTime:
-    now: mock.Mock
+    now: datetime.datetime
     sleep: mock.Mock
+    sleep_async: mock.Mock
 
 
 @pytest.fixture(autouse=True)
```

Comment:
why do you need autouse here? for the new tests you seem to pass this fixture explicitly

Reply:
The `autouse` ensures that, even if a test doesn't ask for `mock_time`, then `NOW_FN`/`SLEEP_FN` still get mocked out! I thought it would help prevent a situation where a test takes a long time because it accidentally indirectly calls the real sleep() function.

Reply:
got ya